### PR TITLE
GEODE-4163 clean up DistributionManager code

### PR DESCRIPTION
**Pull request by bschuchardt**
Removed all dead code - about 800 lines of code have been removed.
Removed DistributionChannel, which served no useful purpose.
Made many methods private or package-private.
Gathered most statics and instance variables scattered around in the file
and moved them to the top.

I attempted to replace use of InternalDistributedMember with
DistributedMember but the use of the former is too widespread.  If we
want to do that it will change a large set of files.

@upthewaterspout @WireBaron @galen-pivotal @PivotalSarge 

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by bschuchardt** [comment]
The PR seems not to have picked up formatting changes that are in 128a6d0d8f8bcbc7a02aee722c42eb2ffa4a169a, causing spotless to lodge a couple of complaints.

File: geode-core/src/main/java/org/apache/geode/distributed/internal/DM.java [hunk 1]
```diff
@@ -461,6 +452,11 @@ void addHostedLocators(InternalDistributedMember member, Collection<String> loca
    */
   void releaseUDPMessagingForCurrentThread();
 
+  /**
+   * returns the type of node
+   * 
+   * @return
```
Comment (upthewaterspout) [inline geode-core/src/main/java/org/apache/geode/distributed/internal/DM.java]: Fill in the @@return to avoid javadoc warnings

**Comment by upthewaterspout** [comment]
The spotless check fails for me with this PR as well, it's not just travis. @bschuchardt maybe you missed pushing some formating changes? Other than that it looks good.

**Comment by bschuchardt** [comment]
I found and fixed the problem with the format violations

**Pull request closed without merging**